toto/toto-run.py

```diff
@@ -19,87 +19,125 @@
   Toto run options are separated from the command to be executed by
   a double dash.
 
+  The implementation of the tasks can be found in runlib.
+
   Example Usage
   ```
-  python -m toto.toto-run --material <files> --product <files> --
-    <command-to-execute> <command-options-and-arguments>
+  ./toto-run.py --step-name write-code --materials . --products . --key bob \
+      -- vi foo.py
   ```
 
-  The actual wrapper and the tasks it performs are implemented in runlib.
-
 """
 
 import os
 import sys
 import argparse
 import toto.util
 import toto.runlib
-
+import toto.log as log
+
+def _die(msg, exitcode=1):
+  log.error(msg)
+  sys.exit(exitcode)
+
+def in_toto_run(step_name, key_path, material_list, product_list,
+    link_cmd_args, record_byproducts=False):
+  """Load link signing private keys from disk and runs passed command, storing
+  its materials, by-products and return value, and products into link metadata
+  file. The link metadata file is signed and stored to disk. """
+  try:
+    log.doing("load link signing key...")
+    key = toto.util.prompt_import_rsa_key_from_file(key_path)
+  except Exception, e:
```

Comment:
I think the `Exception as e` construct is more forward-compatible than this tuple. We may want to verify this.